### ISSUE TITLE
Gradual Unfreezing Question

### ISSUE THREAD

**Issue by DataDaveH** [IndicoDataSolutions/finetune#182]
I am hoping to implement gradual unfreezing while finetuning for an article classification task. I see the config setting called `num_layers_trained`. I thought I could change this setting after each epoch of finetuning, but it seems like the setting is only used during initialization. Is there a recommended way to accomplish this? Thanks!

**Comment by DataDaveH** [comment]
Hey @madisonmay,

Thanks for getting back to me so quickly. I definitely neglected the learning rate scheduling when I tried this, but I did try running 1 epoch at a time and changing `num_layers_trained` between epochs like you said. I couldn't tell if changing `num_layers_trained` between calls to `.fit()` was having any effect. I was expecting to see a difference in train time per epoch when I had say 1 layer unfrozen compared to all 12 layers unfrozen, but the times were the same. Does this seem expected?
Is there a chance that some `model.config` settings that are used at classifier initialization but would have no effect if changed later between calls to `model.fit()`? 
I got a little lost in the code trying to track down where and when these settings are applied. I am going to train all the way through with and without gradual unfreezing if you think my concerns about config settings aren't warranted.

**Comment by madisonmay** [comment]
Sorry for the slow reply, I have been at EMNLP for these past few days and haven't been checking GitHub that actively. Pinging @benleetownsend (the other dev on this project) who might have a better idea what could be going on.

**Comment by benleetownsend** [comment]
@DataDaveH Firstly, apologies for the delay, I did some digging into this today and your intuition was correct and there was a bug with the partial fit logic. After the fix I ran some simple experiments, and I can confirm that it works as intended. The performances were not that great though, so it is likely that you will need to look at learning rate scheduling to get the most out of this.

Keep us in the loop, interested to see what you find.

**Comment by DataDaveH** [comment]
Cool, thanks! Where can I get the fix?

**Comment by benleetownsend** [comment]
Apologies, the fix is in PR #192, I hope to get this merged today but in the meantime you can use the branch associated with that PR.

**Comment by madisonmay** [comment]
Fix has been merged with PR #192, closing!
